compiler-plugin-tests/src/test/resources/ballerina_sources/sample_package_42/service.bal

```diff
@@ -0,0 +1,331 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/time;
+import ballerina/uuid;
+
+type User record {|
+    string id;
+    string name;
+    string email;
+    int age;
+    string department?;
+    string createdAt;
+    string updatedAt;
+|};
+
+type UserInput record {|
+    string name;
+    string email;
+    int age;
+    string department?;
+|};
+
+type UserUpdate record {|
+    string name?;
+    string email?;
+    int age?;
+    string department?;
+|};
+
+type ErrorResponse record {|
+    string message;
+    string code;
+    string timestamp;
+|};
+
+type SuccessResponse record {|
+    string message;
+    json data?;
+|};
+
+type Caller record {|
+    int id;
+|};
+
+map<User> userStore = {};
+
+service /api/v1 on new http:Listener(9090) {
+
+    resource function get users(string? department, int? minAge, int? maxAge) returns User[]|ErrorResponse {
+        User[] users = userStore.toArray();
+        if department is string {
+            users = users.filter(user => user.department == department);
+        }
+        if minAge is int {
+            users = users.filter(user => user.age >= minAge);
+        }
+        if maxAge is int {
+            users = users.filter(user => user.age <= maxAge);
+        }
+        return users;
+    }
+
+    resource function get users/[string id]() returns User|ErrorResponse {
+        User? user = userStore[id];
+        if user is User {
+            return user;
+        } else {
+            return {
+                message: "User not found",
+                code: "USER_NOT_FOUND",
+                timestamp: time:utcToString(time:utcNow())
+            };
+        }
+    }
+
+    resource function get health() returns SuccessResponse {
+        return {
+            message: "Service is healthy",
+            data: {
+                "status": "UP",
+                "timestamp": time:utcToString(time:utcNow()),
+                "totalUsers": userStore.length()
+            }
+        };
+    }
+
+    resource function get stats() returns SuccessResponse {
+        map<int> departmentCount = {};
+        int totalAge = 0;
+
+        foreach User user in userStore {
+            string dept = user.department ?: "Unknown";
+            departmentCount[dept] = (departmentCount[dept] ?: 0) + 1;
+            totalAge += user.age;
+        }
+
+        return {
+            message: "Service statistics",
+            data: {
+                "totalUsers": userStore.length(),
+                "averageAge": userStore.length() > 0 ? totalAge / userStore.length() : 0,
+                "departmentDistribution": departmentCount
+            }
+        };
+    }
+
+    resource function post users(@http:Payload UserInput userInput) returns User|ErrorResponse {
```

Comment:
Shall we remove the payload annotation here and check whether it is treated as the payload parameter?
```suggestion
    resource function post users(UserInput userInput) returns User|ErrorResponse {
```

Reply:
Fixed by [660a2e3](https://github.com/ballerina-platform/module-ballerina-http/pull/2492/commits/660a2e3631d4874079d06e11ffdaad46a28cdcd6)